core/snapshots/storage/bolt.go

```diff
@@ -213,6 +213,37 @@ func GetSnapshot(ctx context.Context, key string) (s Snapshot, err error) {
 	return
 }
 
+// GetAnyStatusSnapshot returns the metadata for any status snapshot transaction
+// referenced by the given key. Requires a context with a storage transaction.
+func GetAnyStatusSnapshot(ctx context.Context, key string) (s Snapshot, err error) {
```

Comment:
This function is almost identical to `GetSnapshot`, except this check:

```go
if s.Kind != snapshots.KindActive && s.Kind != snapshots.KindView {
		return fmt.Errorf("requested snapshot %v not active or view: %w", key, errdefs.ErrFailedPrecondition)
}
```

It should be possible to slightly refactor `GetSnapshot` to accept additional/optional params?

```go
storage.GetSnapshot(ctx, key)
storage.GetSnapshot(ctx, key, WithAnyKind)
```